src/_pytest/terminal.py

```diff
@@ -1636,3 +1638,14 @@ def _get_raw_skip_reason(report: TestReport) -> str:
         elif reason == "Skipped":
             reason = ""
         return reason
+
+
+class _CallableBool:
+    def __init__(self, value: bool):
+        self._value = value
+
+    def __bool__(self):
+        return self._value
+
+    def __call__(self):
+        return self._value
```

Comment:
Adding type annotations:

```suggestion
    def __init__(self, value: bool) -> None:
        self._value = value

    def __bool__(self) -> bool:
        return self._value

    def __call__(self) -> bool:
        return self._value
```